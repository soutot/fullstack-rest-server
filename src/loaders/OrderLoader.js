// @flow
import OrderModel from '../models/order';

export const load = async ({ id }) => {
  const order = await OrderModel.findOne({ id });
  if (!order) return null;

  return order;
};

export const loadAll = async () => {
  const orders = await OrderModel.find();
  return orders;
};

export const edit = async ({ id, values }) => {
  return OrderModel.findOneAndUpdate({ id }, { $set: values }, { runValidators: true, new: true }, (err, doc) => {
    if (err) {
      new Error(`Error: ${err}`);
      return null;
    }
    return doc;
  });
};

const generateId = () => {
  const randomNumber = Math.floor((Math.random() * 1000000) + 1).toString();

  return `SAL${randomNumber.padStart((7), '0')}`;
};

export const add = async ({ values }) => {
  const id = generateId();
  const newOrder = new OrderModel({ ...values, id });
  await newOrder.save(err => {
    if (err) {
      new Error(`Error: ${err}`);
      return null;
    }
  });
  return newOrder;
};
