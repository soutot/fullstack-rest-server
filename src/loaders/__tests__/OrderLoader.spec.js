import * as OrderLoader from '../OrderLoader';

process.env.TEST_SUITE = 'orders-test';

it('should add a new order', async () => {
  const values = {
    price: 10.5,
    customerName: 'Tiago',
    address: {
      street: 'Mocked street',
      number: '10',
      postcode: '098765',
    },
  };

  const order = await OrderLoader.add({ values });

  // TODO: improve this
  const result = {
    ...order._doc,
    _id: 'mocked-order-id',
    address: {
      ...order._doc.address._doc,
      _id: 'mocked-address-id',
    },
  };

  expect(result).toMatchSnapshot();
});
