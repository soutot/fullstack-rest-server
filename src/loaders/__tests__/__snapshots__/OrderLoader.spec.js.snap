// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should add a new order 1`] = `
Object {
  "_id": "mocked-order-_id",
  "address": Object {
    "_id": "mocked-address-id",
    "number": 10,
    "postcode": "098765",
    "street": "Mocked street",
  },
  "customerName": "Tiago",
  "id": "mocked-order-id",
  "price": 10.5,
}
`;

exports[`should edit an order 1`] = `
Object {
  "__v": 0,
  "_id": "mocked-order-_id",
  "address": Object {
    "_id": "mocked-address-id",
    "number": 100,
    "postcode": "123456",
    "street": "Edited Mocked street",
  },
  "customerName": "Tiago",
  "id": "mocked-order-id",
  "price": 10.5,
}
`;
